TYPO3.Neos/Classes/TYPO3/Neos/Service/NodeTypeSchemaBuilder.php

```diff
@@ -75,6 +82,33 @@ public function generateNodeTypeSchema()
     }
 
     /**
+     * @param array $options
+     * @return array
+     * @throws \Exception
+     */
+    protected function loadAlohaPreset(array &$options)
+    {
+        if (array_key_exists('properties', $options)) {
+            foreach ($options['properties'] as &$property) {
+                if (array_key_exists('ui', $property)) {
+                    if (array_key_exists('aloha', $property['ui'])) {
+                        if (array_key_exists('preset', $property['ui']['aloha'])) {
+                            try {
+                                $preset = $this->alohaPresets[$property['ui']['aloha']['preset']];
+                                $originalAlohaConfiguration = $property['ui']['aloha'];
+                                // Merge preset first so a single preset setting can always be overwritten by custom config
+                                $property['ui']['aloha'] = array_merge($preset, $originalAlohaConfiguration);
```

Comment:
Instead of array_merge, shouldn't we rather use `TYPO3\Flow\Utility\Arrays::arrayMergeRecursiveOverrule``?